pkg/helm/release/manager_factory.go

```diff
@@ -63,68 +67,133 @@ func (f managerFactory) NewManager(cr *unstructured.Unstructured) (Manager, erro
+	releaseName, err := getReleaseName(v3StorageBackend, crChart.Name(), cr)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get helm release name: %s", err)
 	}
+	values, ok := cr.Object["spec"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("failed to get spec: expected map[string]interface{}")
+	}
+	controllerRef := metav1.NewControllerRef(cr, cr.GroupVersionKind())
+	ownerRefClient := &ownerRefInjectingClient{
+		refs:   []metav1.OwnerReference{*controllerRef},
+		Client: *kubeClient,
+	}
+	actionConfig := &action.Configuration{
+		RESTClientGetter: rcg,
+		Releases:         v3StorageBackend,
+		KubeClient:       ownerRefClient,
+		Log:              func(_ string, _ ...interface{}) {},
+	}
 	return &manager{
-		storageBackend:   storageBackend,
-		tillerKubeClient: tillerKubeClient,
-		chartDir:         f.chartDir,
+		actionConfig:   actionConfig,
+		storageBackend: v3StorageBackend,
+		kubeClient:     ownerRefClient,
 
-		tiller:      releaseServer,
 		releaseName: releaseName,
 		namespace:   cr.GetNamespace(),
 
-		spec:   cr.Object["spec"],
+		chart:  crChart,
+		values: values,
 		status: types.StatusFor(cr),
 	}, nil
 }
 
-// getReleaseServer creates a ReleaseServer configured with a rendering engine that adds ownerrefs to rendered assets
-// based on the CR.
-func getReleaseServer(cr *unstructured.Unstructured, storageBackend *storage.Storage, tillerKubeClient *kube.Client) (*tiller.ReleaseServer, error) {
-	controllerRef := metav1.NewControllerRef(cr, cr.GroupVersionKind())
-	ownerRefs := []metav1.OwnerReference{
-		*controllerRef,
+type ownerRefInjectingClient struct {
+	refs []metav1.OwnerReference
+	kube.Client
+}
+
+func (c *ownerRefInjectingClient) Build(reader io.Reader, validate bool) (kube.ResourceList, error) {
+	resourceList, err := c.Client.Build(reader, validate)
+	if err != nil {
+		return resourceList, err
 	}
-	baseEngine := helmengine.New()
-	restMapper, err := tillerKubeClient.Factory.ToRESTMapper()
+	err = resourceList.Visit(func(r *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+		objMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(r.Object)
+		if err != nil {
+			return err
+		}
+		u := &unstructured.Unstructured{Object: objMap}
+		if r.ResourceMapping().Scope == meta.RESTScopeNamespace {
+			u.SetOwnerReferences(c.refs)
+		}
+		return nil
+	})
 	if err != nil {
 		return nil, err
 	}
-	e := engine.NewOwnerRefEngine(baseEngine, restMapper, ownerRefs)
-	var ey environment.EngineYard = map[string]environment.Engine{
-		environment.GoTplEngine: e,
+	return resourceList, nil
+}
+
+func convertV2ToV3(v2StorageBackend *v2storage.Storage, v3StorageBackend *v3storage.Storage, cr *unstructured.Unstructured) error {
+	// If a v2 release with the legacy name exists, convert it to v3.
+	legacyName := getLegacyName(cr)
+	legacyHistoryV2, legacyV2Exists, err := releaseHistoryV2(v2StorageBackend, legacyName)
```

Comment:
Nit: some variable/function names have `*V2` or `*V3` name formats, while others have `v2*` or `v3*`; it would be nice to give all names the same format.

Reply:
I think I've got them all covered. Let me know if you see anymore I missed.